vite-project-kabinet/src/Content/CardPreview.jsx

```diff
@@ -0,0 +1,37 @@
+import { Image } from "./Image.jsx"
+import { CardPreface } from "./CardPreface.jsx"
+import { Link } from 'react-router';
+import "./card.style.css"
+
+export const CardPreview = ({genre, title, author, content, image}) => {
+
+    const cardClassName = () => {
+        if (image) {
+            if (genre === "esej") {
+                return `card card--esej card--image`
+            }
+            else {
+                return `card card--texty card--image`
+            }
+        } 
+        else {
+            if (genre === "esej") {
+                return `card card--esej`
+            }
+            else {
+                return `card card--texty`
+            }
+        }
+    }
```

Comment:
Tohle jde napsat trochu lépe, tady se Ti opakují třídy, takže případné budoucí změny budou náchylnější k chybám. Tohle by mělo být víc future-proof (nezkoušel jsem, tak nevím jestli tam nemá překlep nebo tak)

```suggestion
const classes = ["card"]
            if (genre === "esej") {
               classes.push("card-esej");
            }
            else {
            classes.push("card-texty");
            }
            if (image) {
               classes.push("card--image");
              }
     return classes.join(" ");
    }
```

Reply:
Díky, prepísala som to v komponente a premenovala triedy na card--texts a card--essay.